core/src/test/java/org/apache/iceberg/TestManifestWriterVersions.java

```diff
@@ -213,27 +228,125 @@ public void testV2ManifestRewriteWithInheritance() throws IOException {
     checkRewrittenEntry(readManifest(manifest2), 0L, FileContent.DATA);
   }
 
+  @Test
+  public void testV3Write() throws IOException {
+    ManifestFile manifest = writeManifest(3);
+    checkEntry(
+        readManifest(manifest),
+        ManifestWriter.UNASSIGNED_SEQ,
+        ManifestWriter.UNASSIGNED_SEQ,
+        FileContent.DATA,
+        FIRST_ROW_ID);
+  }
+
+  @Test
+  public void testV3WriteWithInheritance() throws IOException {
```

Comment:
```suggestion
 public void testV3WriteWithRowIdInheritance() throws IOException {
```

Reply:
Hmm, I guess it's named because we're check _both_ sequence and row id inheritance.

Reply:
This tests v2 inheritance as well and adds row IDs to the existing checks.